github/repos.go

```diff
@@ -727,6 +727,9 @@ type Protection struct {
 	RequiredPullRequestReviews *PullRequestReviewsEnforcement `json:"required_pull_request_reviews"`
 	EnforceAdmins              *AdminEnforcement              `json:"enforce_admins"`
 	Restrictions               *BranchRestrictions            `json:"restrictions"`
+	RequireLinearHistory       *LinearHistoryEnforcement      `json:"required_linear_history"`
+	AllowForcePushes           *ForcePushTolerance            `json:"allow_force_pushes"`
+	AllowDeletions             *DeletionTolerance             `json:"allow_deletions"`
```

Comment:
Although these three new struct names are good choices on their own, let's please rename them to match the name of the field, for two reasons... first, to be more consistent with the rest of the repo, and second to reduce confusion when searching for the struct or the field.  So to be clear, let's please rename these structs to `RequireLinearHistory`, `AllowForcePushes`, and `AllowDeletions`, respectively.